### PR TITLE
add DELETE /api/comments/:comment_id endpoint

### PR DESCRIPTION
**Pull request by dku7**
Add ability to delete the comment specified by `comment_id`.

- Tested the following:
  -  `204` - successfully delete the comment
  -  `400` - "Bad request" when the specified `comment_id` is not the correct data type
  - `404` - "Not found" when the specified `comment_id` is not found in database

- Added corresponding `controller` and `model` functions.
- Updated `endpoints.json` with details for the endpoint.

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -321,3 +321,32 @@ describe("/api/articles/:article_id/comments", () => {
...
+    test("DELETE: 204 - respond with status code of 204 with no content on successfully deleting a comment", () => {
+      const comment_id = 5;
+      return request(app)
+        .delete(`/api/comments/${comment_id}`)
+        .expect(204)
+        .then(() =>
+          // request GET to check we get a 404 after deleting
+          request(app).get(`/api/comments/${comment_id}`).expect(404)
+        );
+    });
```
Comment (nc-jodie-cheung) [inline __tests__/app.test.js:336]: You are not making a GET request here, you are deleting. Your test should be like this:
` request(app).delete(`/api/comments/${comment_id}`).expect(204)`

File: controllers/comments.controller.js [hunk 2]
```diff
@@ -28,4 +30,19 @@ function postNewComment(request, response, next) {
...
+
+function removeCommentByCommentId(request, response, next) {
+  const { comment_id } = request.params;
+
+  // check the comment exists first
+  return selectCommentByCommentId(comment_id)
+    .then(() => deleteCommentByCommentId(comment_id))
+    .then(() => response.status(204).send())
+    .catch(next);
+}
```
Comment (nc-jodie-cheung) [inline controllers/comments.controller.js:42]: You can simplify your code by removing the extra step of checking if the comment exists before deleting it. The database will handle this for you 😄

File: __tests__/app.test.js (hunk 1 above)
Comment (dku7) [reply __tests__/app.test.js:336]: Resolved in https://github.com/dku7/be-nc-news/pull/19

File: controllers/comments.controller.js (hunk 2 above)
Comment (dku7) [reply controllers/comments.controller.js:42]: Resolved in https://github.com/dku7/be-nc-news/pull/19

**Pull request merged**